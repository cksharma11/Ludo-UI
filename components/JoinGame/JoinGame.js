import React from 'react';
import Header from '../Header/Header';
import JoinGameStyles from './JoinGame.style';
import labels from '../../config/labels/labels';

const JoinGame = () => {
  return (
    <div className="container">
      <Header title="Ludo" />
      <div className="form-container">
        <form method="POST" action="/joinGame">
          <div className="activity-header">{labels.JOIN_GAME}</div>
          <input type="text" name="playerName" placeholder="name" />
          <input type="number" name="gameId" placeholder="gameId" />
          <input type="submit" value={labels.JOIN_GAME} />
        </form>
      </div>
      <style jsx>{JoinGameStyles}</style>
    </div>
  );
};

export default JoinGame;
