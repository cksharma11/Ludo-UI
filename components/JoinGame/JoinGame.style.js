import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const JoinGameStyles = css`
  form {
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
      'Lucida Sans', Arial, sans-serif;
    display: flex;
    height: 660px;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: ${colors.palette1.shade2};
  }

  form > input {
    background: ${colors.gray.shade3};
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
      'Lucida Sans', Arial, sans-serif;
    width: 200px;
    height: 30px;
    border: 1px solid ${colors.gray.shade1};
    color: black;
    text-align: center;
    line-height: 2;
    text-decoration: none;
    padding: 5px;
    margin: 5px;
    border-radius: 2px;
  }

  .activity-header {
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
      'Lucida Sans', Arial, sans-serif;
    font-size: 30px;
  }
`;

export default JoinGameStyles;
