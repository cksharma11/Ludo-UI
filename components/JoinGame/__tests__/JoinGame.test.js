import React from 'react';
import { shallow } from 'enzyme';
import JoinGame from '../JoinGame';

describe('JoinGame', () => {
  it('should render properly', () => {
    const wrapper = shallow(<JoinGame />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
