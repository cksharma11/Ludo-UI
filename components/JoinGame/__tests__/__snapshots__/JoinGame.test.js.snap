// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinGame should render properly 1`] = `
"<Fragment>
  <Header title=\\"Ludo\\" />
  <form method=\\"POST\\" action=\\"/joinGame\\" className=\\"jsx-1016620996\\">
    <div className=\\"jsx-1016620996 activity-header\\">
      Join Game
    </div>
    <input type=\\"text\\" name=\\"playerName\\" placeholder=\\"name\\" className=\\"jsx-1016620996\\" />
    <input type=\\"number\\" name=\\"gameId\\" placeholder=\\"gameId\\" className=\\"jsx-1016620996\\" />
    <input type=\\"submit\\" value=\\"Join Game\\" className=\\"jsx-1016620996\\" />
  </form>
  <JSXStyle id=\\"1016620996\\">
    f
    o
    r
    m
    .
    j
    s
    x
    -
    1
    0
    1
    6
    6
    2
    0
    9
    9
    6
    {
    f
    o
    n
    t
    -
    f
    a
    m
    i
    l
    y
    :
    &#39;
    T
    r
    e
    b
    u
    c
    h
    e
    t
     
    M
    S
    &#39;
    ,
    &#39;
    L
    u
    c
    i
    d
    a
     
    S
    a
    n
    s
     
    U
    n
    i
    c
    o
    d
    e
    &#39;
    ,
    &#39;
    L
    u
    c
    i
    d
    a
     
    G
    r
    a
    n
    d
    e
    &#39;
    ,
     
    &#39;
    L
    u
    c
    i
    d
    a
     
    S
    a
    n
    s
    &#39;
    ,
    A
    r
    i
    a
    l
    ,
    s
    a
    n
    s
    -
    s
    e
    r
    i
    f
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    6
    6
    0
    p
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    d
    f
    d
    f
    d
    f
    ;
    }
    f
    o
    r
    m
    .
    j
    s
    x
    -
    1
    0
    1
    6
    6
    2
    0
    9
    9
    6
    &gt;
    i
    n
    p
    u
    t
    .
    j
    s
    x
    -
    1
    0
    1
    6
    6
    2
    0
    9
    9
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    e
    e
    e
    e
    e
    e
    ;
    f
    o
    n
    t
    -
    f
    a
    m
    i
    l
    y
    :
    &#39;
    T
    r
    e
    b
    u
    c
    h
    e
    t
     
    M
    S
    &#39;
    ,
    &#39;
    L
    u
    c
    i
    d
    a
     
    S
    a
    n
    s
     
    U
    n
    i
    c
    o
    d
    e
    &#39;
    ,
    &#39;
    L
    u
    c
    i
    d
    a
     
    G
    r
    a
    n
    d
    e
    &#39;
    ,
     
    &#39;
    L
    u
    c
    i
    d
    a
     
    S
    a
    n
    s
    &#39;
    ,
    A
    r
    i
    a
    l
    ,
    s
    a
    n
    s
    -
    s
    e
    r
    i
    f
    ;
    w
    i
    d
    t
    h
    :
    2
    0
    0
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    3
    0
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    2
    2
    2
    8
    3
    1
    ;
    c
    o
    l
    o
    r
    :
    b
    l
    a
    c
    k
    ;
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    l
    i
    n
    e
    -
    h
    e
    i
    g
    h
    t
    :
    2
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    t
    e
    x
    t
    -
    d
    e
    c
    o
    r
    a
    t
    i
    o
    n
    :
    n
    o
    n
    e
    ;
    t
    e
    x
    t
    -
    d
    e
    c
    o
    r
    a
    t
    i
    o
    n
    :
    n
    o
    n
    e
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    5
    p
    x
    ;
    m
    a
    r
    g
    i
    n
    :
    5
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    2
    p
    x
    ;
    }
    .
    a
    c
    t
    i
    v
    i
    t
    y
    -
    h
    e
    a
    d
    e
    r
    .
    j
    s
    x
    -
    1
    0
    1
    6
    6
    2
    0
    9
    9
    6
    {
    f
    o
    n
    t
    -
    f
    a
    m
    i
    l
    y
    :
    &#39;
    T
    r
    e
    b
    u
    c
    h
    e
    t
     
    M
    S
    &#39;
    ,
    &#39;
    L
    u
    c
    i
    d
    a
     
    S
    a
    n
    s
     
    U
    n
    i
    c
    o
    d
    e
    &#39;
    ,
    &#39;
    L
    u
    c
    i
    d
    a
     
    G
    r
    a
    n
    d
    e
    &#39;
    ,
     
    &#39;
    L
    u
    c
    i
    d
    a
     
    S
    a
    n
    s
    &#39;
    ,
    A
    r
    i
    a
    l
    ,
    s
    a
    n
    s
    -
    s
    e
    r
    i
    f
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    3
    0
    p
    x
    ;
    }
  </JSXStyle>
</Fragment>"
`;
